### PR TITLE
Adding sub total and sorting capabilities

### PR DESCRIPTION
**Pull request by slevertbiot**

### CODE REVIEW

File: udmd/api/src/device/dao/mongodb/MongoDeviceDAO.ts [hunk 1]
```diff
@@ -20,11 +20,16 @@ export class MongoDeviceDAO implements DeviceDAO {
       .toArray();
   }
 
+  async getFilteredDeviceCount(searchOptions: SearchOptions): Promise<number> {
+    // console.log(await this.db.collection<Device>('device').countDocuments());
```
Comment (slevertbiot) [inline udmd/api/src/device/dao/mongodb/MongoDeviceDAO.ts:24]: will remove this

File: udmd/api/src/device/dao/mongodb/MongoFilterBuilder.ts [hunk 2]
```diff
@@ -5,11 +5,16 @@ export function getFilter(jsonFilters: Filter[]): any {
   const mongoFilters: any = {};
 
   jsonFilters.forEach((filter) => {
+    if (filter.field === 'operational') {
+    }
     // '~' is our symbol for 'contains'
     // the operational field is a boolean so do not try a partial match on this field
```
Comment (slevertbiot) [inline udmd/api/src/device/dao/mongodb/MongoFilterBuilder.ts:11]: going to clean up these docs

File: udmd/api/src/__tests__/device/dao/mongo/MongoDeviceDAO.spec.ts [hunk 3]
```diff
@@ -41,8 +42,9 @@ beforeEach(async () => {
 
 afterEach(async () => {
   await db.collection('device').deleteMany({});
-  jest.resetAllMocks();
-  jest.restoreAllMocks();
+
+  // jest.resetAllMocks();
```
Comment (mspratt-biot) [inline udmd/api/src/__tests__/device/dao/mongo/MongoDeviceDAO.spec.ts:46]: Can remove comments.

**Review by mspratt-biot — APPROVED** [review]

File: udmd/api/src/__tests__/device/dao/mongo/MongoDeviceDAO.spec.ts [hunk 4]
```diff
@@ -43,8 +45,8 @@ beforeEach(async () => {
 afterEach(async () => {
   await db.collection('device').deleteMany({});
 
-  // jest.resetAllMocks();
-  // jest.restoreAllMocks();
+  jest.resetAllMocks();
```
Comment (mspratt-biot) [inline udmd/api/src/__tests__/device/dao/mongo/MongoDeviceDAO.spec.ts:48]: u ended up needing these?

**Review by mspratt-biot — APPROVED** [review]

**Pull request merged**